### PR TITLE
Fabo/1189-catch reconnect errors

### LINKED ISSUE #1189: Surface errors from the main thread better
UI Version: 0.9.4

Description:

`main-process: gaiacli: ERROR: unknown command "rest-server" for "gaiacli"` did not surface to the UI.

### PR DESCRIPTION
**Pull request by faboweb**
Closes #1189 

Description:
the async call for reconnection wasn't handling errors. now in the connect function handles errors directly.

❤️ Thank you!

### CODE REVIEW

File: app/src/main/index.js [hunk 1]
```diff
@@ -98,7 +98,11 @@ function handleCrash(error) {
   afterBooted(() => {
     if (mainWindow) {
       mainWindow.webContents.send("error", {
-        message: error ? error.message : undefined
+        message: error
+          ? error.message
+            ? error.message
+            : error
+          : "An unspecified error occurred"
```
Comment (jbibla) [inline app/src/main/index.js]: can we give any more info?

File: app/src/main/index.js [hunk 2]
```diff
@@ -269,6 +274,13 @@ async function startLCD(home, nodeIP) {
     ])
     logProcess(child, join(home, "lcd.log"))
 
+    child.on("exit", () => {
+      let msg = `The ${LCD_BINARY_NAME} rest-server (LCD) exited unplanned`
```
Comment (jbibla) [inline app/src/main/index.js]: same here?

File: app/src/main/index.js (hunk 1 above)
Comment (fedekunze) [reply app/src/main/index.js]: ++

Comment (faboweb) [reply app/src/main/index.js]: If there is no error, we can not guess the error. ;)
We can change the error message to sth. More meaningful maybe? Do u guys have sth in mind?

Comment (jbibla) [reply app/src/main/index.js]: what are the circumstances where this could happen?

Comment (faboweb) [reply app/src/main/index.js]: i.e. an uncaught `Promise.reject` without error message. We should avoid having those, but still have the fallback if we forget it.

**Pull request merged**